utils.go

```diff
@@ -13,3 +13,36 @@ func subSlice[S ~[]E, E any](s S, i, j int) S {
 	}
 	return s[i:j]
 }
+
+// Find returns the smallest index i at which x == a[i],
+// or len(a) if there is no such index.
+func find(a []string, x string) int {
+	for i, n := range a {
+		if x == n {
+			return i
+		}
+	}
+	return -1
+}
+
+// Contains tells whether a contains x.
+func contains(a []string, x string) bool {
+	for _, n := range a {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+// Remove by name
```

Comment:
According to impl, suggest update doc to "Remove the first apperence of item in slice l."
Also I'd prefer to remove the deug lines in the impl.

Reply:
ok, corrected
I am also not leaving debug lines, I am always doing cleanups for final PR.